crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1208,6 +1208,13 @@ impl Client {
         Ok(Room::new(self.clone(), base_room))
     }
 
+    /// Leave a room given its `RoomId`.
```

Comment:
We don't need this method, it's the same as `Room::leave()`, and we can only leave a room we do know about (i.e. we can get it with `Client::get_room` first).

Reply:
I think we're going in circles... We don't have anything in the `ffi::Client` to get a `Room`, and even if we had it, it would be a 'not safe to use' Room as we had in `RoomListItem::invited_room` (see https://github.com/matrix-org/matrix-rust-sdk/pull/4152#discussion_r1808445972). That's the main reason I split Room and RoomPreview here, and I think it's what we agreed to do.

So now I'm not sure what to do anymore:
- If we're ok with returning a Room with no timeline as we did before then I'll just remove everything from `RoomPreview` and just return that Room in `RoomListItem::preview_room` instead.
- If we're not ok with that, then we need a way to be able to leave that room (aka decline the invite, cancel the knock).